onnxruntime/test/python/onnxruntime_test_python.py

```diff
@@ -91,17 +106,40 @@ def testSetProvidersWithOptions(self):
             option['trt_int8_calibration_table_name'] = calib_table_name
             int8_use_native_calibration_table = "true"
             option['trt_int8_use_native_calibration_table'] = int8_use_native_calibration_table
+            dla_enable = "true"
+            option['trt_dla_enable'] = dla_enable
+            option['trt_dla_core'] = new_dla_core
+            dump_subgraphs = "true"
+            option['trt_dump_subgraphs'] = dump_subgraphs
+            engine_cache_enable = "true"
+            option['trt_engine_cache_enable'] = engine_cache_enable
+            engine_cache_path = '/home/onnxruntime/engine_cache'
+            option['trt_engine_cache_path'] = engine_cache_path
+            engine_decryption_enable = "true"
+            option['trt_engine_decryption_enable'] = engine_decryption_enable
+            engine_decryption_lib_path = '/home/onnxruntime/decryption_lib'
+            option['trt_engine_decryption_lib_path'] = engine_decryption_lib_path
+            force_sequential_engine_build = "true"
+            option['trt_force_sequential_engine_build'] = force_sequential_engine_build
             sess.set_providers(['TensorrtExecutionProvider'], [option])
 
             options = sess.get_provider_options()
             option = options['TensorrtExecutionProvider']
+            self.assertEqual(option['trt_max_partition_iterations'], str(new_max_partition_iterations))
+            self.assertEqual(option['trt_min_subgraph_size'], str(new_min_subgraph_size))
             self.assertEqual(option['trt_max_workspace_size'], str(new_max_workspace_size))
             self.assertEqual(option['trt_int8_calibration_table_name'], str(calib_table_name))
-            self.assertEqual(option['has_trt_options'], '1')
             self.assertEqual(option['trt_fp16_enable'], '1')
             self.assertEqual(option['trt_int8_enable'], '0')
             self.assertEqual(option['trt_int8_use_native_calibration_table'], '1')
-
+            self.assertEqual(option['trt_dla_enable'], '1')
+            self.assertEqual(option['trt_dla_core'], str(new_dla_core))
+            self.assertEqual(option['trt_dump_subgraphs'], '1')
+            self.assertEqual(option['trt_engine_cache_enable'], '1')
```

Comment:
for some of these options, it would be better to see if it actually worked.
e.g. one test could be if we enable fp16 and engine_cache , let's see if an engine cache file (with fp16 in the name) was actually created.